metaflow/plugins/kfp/s3_sensor_decorator.py

```diff
@@ -0,0 +1,43 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+
+from types import FunctionType
+from typing import Tuple
+
+def identity_formatter(path: str, flow_parameters_json: dict) -> str:
+    return key
+
+def split_s3_path(path: str) -> Tuple[str, str]:
+    path = path.replace("s3://", "")
+    bucket, key = path.split("/", 1)
+    return bucket, key
+
+class S3SensorDecorator(FlowDecorator):
+    name = 's3_sensor'
+    defaults = {
+        "path": "",
```

Comment:
What variables does a path format string have access to? 
It should have access to any variable in `flow_parameters_json`, and some more like `date,year,day,hour` etc.
```python
f"$date
```

How about access to environment variables?

Reply:
It currently has access to any variable in `flow_parameters_json`, which is why the formatter function looks like `def formatter(key: str, flow_parameters_json: dict) -> str:`. I'll add support for the `date,year,day,hour` shortly, but haven't yet confirmed if environment variables are available.

Reply:
In terms of environment variables, the user has access to these default env vars available in the pods (I verified):
```
        - name: MF_POD_NAME
        - name: MF_POD_NAMESPACE
        - name: MF_ARGO_NODE_NAME
        - name: MF_ARGO_WORKFLOW_NAME
        - name: ZODIAC_SERVICE
        - name: ZODIAC_TEAM
        - name: POD_NAMESPACE
        - name: KFP_SDK_NAMESPACE
        - name: POD_NAME
        - name: POD_IP
        - name: ARGO_WORKFLOW_NAME
        - name: K8S_CLUSTER_NAME
        - name: K8S_CLUSTER_ENV
        - name: METAFLOW_NOTIFY_EMAIL_SMTP_HOST
        - name: METAFLOW_NOTIFY_EMAIL_SMTP_PORT
        - name: METAFLOW_NOTIFY_EMAIL_FROM
        - name: METAFLOW_NOTIFY_EMAIL_BODY
        - name: METAFLOW_NOTIFY_ON_ERROR
        - name: KFP_RUN_URL_PREFIX
        - name: METAFLOW_DEFAULT_DATASTORE
        - name: METAFLOW_DEFAULT_METADATA
        - name: METAFLOW_SERVICE_URL
        - name: METAFLOW_DATASTORE_SYSROOT_S3
        - name: WAGGLE_DANCE_DEFAULT_ENVIRONMENT
        - name: WAGGLE_DANCE_DEV_ENDPOINT
        - name: WAGGLE_DANCE_STAGE_ENDPOINT
        - name: WAGGLE_DANCE_PROD_READ_ONLY_ENDPOINT
        - name: WAGGLE_DANCE_PROD_ENDPOINT
        - name: AWS_ROLE_ARN
        - name: AWS_WEB_IDENTITY_TOKEN_FILE
```

Reply:
@talebzeghmi Alex's comments at the bottom [here](https://docs.google.com/document/d/1_3ZpLu323X0Wc-grvsHbee2ZnND-uSX3vCw4Q64H_UQ/edit?disco=AAAANNFnu3Q) make a lot of sense, so I'd say we not introduce any addition functionality like `$date`, `$year`, etc and keep it generic for customers.

Even if we were to provide those values, the only way to use something like `datetime.now()` as we can't assume how customers will name their `scheduled_time` parameter, and `datetime.now()` obviously won't work well in production (due to the fact that a flow might get scheduled late).

Reply:
What values do you provide? Can you please document them? Let's use them also from within a test flow

Reply:
We won't provide any values other than what's already available in `flow_parameters_json` + the environment variables . But I don't expect customers to find much value in the environment variables, except perhaps `ZODIAC_SERVICE` and `ZODIAC_NAME`. I can update the test to include an environment variable.

Reply:
Added documentation and updated test.